backend/app/analysis/dependency_parsing.py

```diff
@@ -0,0 +1,218 @@
+import os
```

Comment:
This module requires `gender_analysis`, but it's not added to the project requirements file. If we want to make `gender_analysis` a dependency of the project, why not do so and import this module directly from it?